validate.go

```diff
@@ -29,10 +29,10 @@ func adjustResourceRequest(container *corev1.Container, resourceName string, res
 // Return `true` when the container has been mutated
 func validateAndAdjustContainerResourceRequests(container *corev1.Container, settings *Settings) bool {
 	mutated := false
-	if settings.Memory != nil {
+	if !settings.Memory.IgnoreValues && settings.Memory != nil {
 		mutated = adjustResourceRequest(container, "memory", settings.Memory)
 	}
-	if settings.Cpu != nil {
+	if !settings.Memory.IgnoreValues && settings.Cpu != nil {
```

Comment:
Wrong, it should be `settings.Cpu.IgnoreValues`, before fixing it, can you reproduce the issue with a unit test?

Reply:
Furthermore, I think you need to invert the && values. If the `settings.Cpu` is nil, the program will panic.

Reply:
I've done a slight refoctor that resolves this issue and removes this condition on this IF.